torchrl/objectives/sac.py

```diff
@@ -831,8 +944,44 @@ def _forward_value_estimator_keys(self, **kwargs) -> None:
         if self._value_estimator is not None:
             self._value_estimator.set_keys(
                 value=self._tensor_keys.value,
+                reward=self.tensor_keys.reward,
+                done=self.tensor_keys.done,
             )
 
+    @property
+    def in_keys(self):
+        keys = [
+            self.tensor_keys.action,
+            ("next", self.tensor_keys.reward),
+            ("next", self.tensor_keys.done),
+            *self.actor_network.in_keys,
+            *[("next", key) for key in self.actor_network.in_keys],
+            *self.qvalue_network.in_keys,
+        ]
+        return list(set(keys))
```

Comment:
just realizing that
this is pretty slow, and in_keys will be called often. I would memoize that with an `_in_keys` like we do for `out_keys`

Reply:
Good point, I will port this back to the other `in_keys` of the loss modules. 
It's worth noting, however, that each time the .set_keys() method is invoked, we will also need to update the _in_keys accordingly.

Reply:
we could have `set_keys()` destroy `_in_keys`